dags/remote_automated_ampliseq.py

```diff
@@ -0,0 +1,373 @@
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def monitor_status_nextflow(study):
+            logging.info(f"Monitoring Nextflow job status for study {study['study']}...")
+            job_sensor = JobStatusSensor(
+                task_id=f"monitor_status_nextflow_{study['study']}",
+                remote_host=REMOTE_HOST_SSH,
+                username=USERNAME,
+                key_file=KEY_FILE,
+                poke_interval=POKE_INTERVAL,
+            )
+            return job_sensor.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def compress_remote_results(study):
+            logging.info(f"Compressing remote results for study {study['study']}...")
+            command = f'tar -czf {REMOTE_PATH}/{study["study"]}_results.tar.gz -C {REMOTE_PATH}/{study["study"]} out'
+            ssh_op = CustomSSHOperator(
+                task_id='compress_remote_results_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SSH,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                command=command
+            )
+            ssh_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def transfer_compressed_results(study):
+            logging.info(f"Transferring compressed results for study {study['study']}...")
+            remote_compressed_path = os.path.join(REMOTE_PATH, f"{study['study']}_results.tar.gz")
+            local_compressed_path = os.path.join(study['path'], f"{study['study']}_results.tar.gz")
+            sftp_op = SFTPOperator(
+                task_id='transfer_compressed_results_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SFTP,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                local_filepath=local_compressed_path,
+                remote_filepath=remote_compressed_path,
+                operation='get',
+                create_intermediate_dirs=True
+            )
+            sftp_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def extract_results_file(study):
+            logging.info(f"Extracting results file for study {study['study']}...")
+            local_compressed_path = os.path.join(study['path'], f"{study['study']}_results.tar.gz")
+            local_extract_path = study['path']
+            with tarfile.open(local_compressed_path, 'r:gz') as tar:
+                tar.extractall(path=local_extract_path)
+            os.remove(local_compressed_path)
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def local_Rscript_postprocessing(study):
+            logging.info(f"Running local Rscript postprocessing for study {study['study']}...")
+            command = f'Rscript {BASE_PATH}/ampliseq_postProcessing.R {study["study"]} {study["path"]}/out'
```

Comment:
question of general curiosity thats just occurred to me.. are you familiar w R? it would maybe be good to be able to debug these ..